src/nvim/main.c

```diff
@@ -1926,22 +1924,26 @@ static bool file_owned(const char *fname)
 #endif
 
 /// Prints the following then exits:
-/// - An error message main_errors[n]
-/// - A string str if not null
+/// - An error message `errstr`
+/// - A string `str` if not null
 ///
-/// @param n    error number represented by an ME_* macro
-/// @param str  string to append to the primary error message, or NULL
-static void mainerr(int n, const char *str)
+/// @param errstr  string containing an error message
+/// @param str     string to append to the primary error message, or NULL
+static void mainerr(const char *errstr, const char *str)
 {
   signal_stop();              // kill us with CTRL-C here, if you like
 
-  mch_errmsg(_(main_errors[n]));
+  mch_errmsg(argv0);
+  mch_errmsg(": ");
+  mch_errmsg(_(errstr));
   if (str != NULL) {
     mch_errmsg(": \"");
     mch_errmsg(str);
     mch_errmsg("\"");
   }
-  mch_errmsg(_("\nMore info with \"nvim -h\"\n"));
+  mch_errmsg(_("\nMore info with \""));
+  mch_errmsg(argv0);
+  mch_errmsg(_(" -h\"\n"));
```

Comment:
No, I suppose we don't. The case statement dealing with `-h` just takes an ASCII character.